pkg/agent/openflow/feature_templates.go

```diff
@@ -0,0 +1,369 @@
+//      |               Stage                |        |     Stage    |       |        Stage       |
+//      +------------------------------------+        +--------------+       +--------------------+
+//              \                                          /    \                        /
+//               \                                        /      \                      /
+//                \                                      /        \                    /
+//              +------------------------------------------+    +--------------------------+
+//              |                 Feature                  |    |          Feature         |
+//              +------------------------------------------+    +--------------------------+
+//                                  |                    \                         /
+//                                  |                     \                       /
+//                                  |                      \                     /
+//                     +------------------------+        +-------------------------+
+//                     |        Pipeline        |        |         Pipeline        |
+//                     +------------------------+        +-------------------------+
+//
+// FeatureTable:
+// FeatureTable in Flexible Pipeline is like "iptables entry" in iptables. It is the basic unit to build OVS pipelines.
+//
+// Stage:
+// Stage in Flexible Pipeline is like "chain" in iptables. The order of Stages is fixed. A Stage can be empty or have
+// one or more FeatureTables. Note that, a FeatureTable cannot be shared by different Stages. All Stages include,
+//   - ClassifierStage
+//	 - ValidationStage
+//	 - ConntrackStateStage
+//	 - PreRoutingStage
+//	 - EgressSecurityStage
+//	 - RoutingStage
+//	 - PostRoutingStage
+//	 - SwitchingStage
+//	 - IngressSecurityStage
+//	 - ConntrackStage
+//	 - OutputStage
+//
+// Feature:
+// Feature in Flexible Pipeline is like "table" in iptables. A Feature is intended to implement a major function. Note
+// that, a Feature can have one or more Stages, and a Stage can be shared by multiple Features. All Features include,
+//   - PodConnectivity, implementation of connectivity for Pods
+//   - NetworkPolicy, implementation of K8s NetworkPolicy and Antrea NetworkPolicy
+//   - Service, implementation of K8s Service
+//   - Egress, implementation of Egress
+//   - Traceflow, implementation of Traceflow
+//   - Multicast, implementation of Multicast
+//
+// Pipeline:
+// This is a concept in Flexible Pipeline, not OVS pipeline, but a corresponding OVS pipeline will be realized in OVS
+// finally. A Pipeline is intended to implement a set of major functions. Note that, a Pipeline can have one or more
+// Features, but a Feature can be shared by multiple Pipelines. All Pipelines include,
+//   - Pipeline for IP
+//   - Pipeline for ARP
+//   - Pipeline for multicast
+//
+// How to insert a flow table with framework Flexible Pipeline?
```

Comment:
Can we also add a few description about how to add a Feature in Flexible Pipeline?

Reply:
Agreed, thanks for reminding.